src/Draco.LanguageServer/Handlers/DracoFindAllReferencesHandler.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Draco.Compiler.Api;
+using Draco.Compiler.Api.Syntax;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Draco.LanguageServer.Handlers;
+internal sealed class DracoFindAllReferencesHandler : ReferencesHandlerBase
+{
+    private readonly DracoDocumentRepository documentRepository;
+
+    public DracoFindAllReferencesHandler(DracoDocumentRepository documentRepository)
+    {
+        this.documentRepository = documentRepository;
+    }
+
+    protected override ReferenceRegistrationOptions CreateRegistrationOptions(
+        ReferenceCapability capability,
+        ClientCapabilities clientCapabilities) => new()
+        {
+            DocumentSelector = new DocumentSelector(new DocumentFilter
+            {
+                Pattern = $"**/*{Constants.DracoSourceExtension}",
+            })
+        };
```

Comment:
We might need to factor out the document selector as a whole, pops up all around